prow/github/types.go

```diff
@@ -1090,20 +1094,70 @@ type Milestone struct {
 // Note that it's wrapping a GitCommit, so author/committer information is in two places,
 // but contain different details about them: in RepositoryCommit "github details", in GitCommit - "git details".
 type RepositoryCommit struct {
-	SHA         string    `json:"sha"`
-	Commit      GitCommit `json:"commit"`
-	Author      User      `json:"author"`
-	Committer   User      `json:"committer"`
-	Parents     []Commit  `json:"parents,omitempty"`
-	HTMLURL     string    `json:"html_url"`
-	URL         string    `json:"url"`
-	CommentsURL string    `json:"comments_url"`
+	NodeID      string      `json:"node_id,omitempty"`
```

Comment:
@hi-rustin please scope this PR back down what you initially had, extending the `RepositoryCommit` structs and its elements. If you want to do further changes to the types here, you can create follow-ups but then:
* Provide a reference to the github api docs for the relevant struct and add it as a godoc comment, if applicable
* Provide a sample of where this struct is used, so it becomes clear that its actually used for the referenced api endpoint

Changes in the gh types like this have a huge blast radius, so if you want a chance at getting them merged you need to:
* Keep them well-scoped
* Convince me that this change is safe (the ever-expanding scope of this PR combined with the lack of references is the main reason its sitting so long, going through our codebase and the github api docs in order to determine if a change I personally don't need is safe is very low on my priority list)

Reply:
Reverted. How about now?

Reply:
Now all structs copy from go-gihub.